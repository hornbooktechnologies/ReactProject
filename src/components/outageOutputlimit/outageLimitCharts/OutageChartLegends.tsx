import { useTranslation } from "react-i18next";

const OutageChartLegends = () => {
  const { i18n } = useTranslation()
  return (
    <div className="timeseries-legends">
      <div className="uk-flex uk-flex-middle uk-flex-center" uk-flex="">
        <div className="uk-flex uk-flex-middle pl-20 mb-10 ml-10 mr-10">
          <span className="indicate actualLegend"></span>
          <div className="ml-10">
            <p className="uk-margin-remove font-12">
              {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.CHART_LEGENDS.ACTUAL')}
            </p>
          </div>
        </div>
        <div className="uk-flex uk-flex-middle pl-20 mb-10 ml-10 mr-10">
          <span className="indicate forecastLegend"></span>
          <div className="ml-10">
            <p className="uk-margin-remove font-12">
              {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.CHART_LEGENDS.FORECAST')}
            </p>
          </div>
        </div>
        <div className="uk-flex uk-flex-middle pl-20 mb-10 ml-10 mr-10">
          <span className="indicate salesLegend"></span>
          <div className="ml-10">
            <p className="uk-margin-remove font-12">
              {i18n.t(
                'OUTAGE_AND_OUTPUT_LIMIT.CHART_LEGENDS.SALES_UNIT_PRICE'
              )}
            </p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default OutageChartLegends;