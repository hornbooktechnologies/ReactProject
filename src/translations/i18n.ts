import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
