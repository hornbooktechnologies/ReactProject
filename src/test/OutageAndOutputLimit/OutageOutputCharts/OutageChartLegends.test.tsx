import React from "react";
import OutageChartLegends from "../../../components/outageOutputlimit/outageLimitCharts/OutageChartLegends";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: OutageChartLegends", () => {
  it("OutageChartLegends renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <OutageChartLegends />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
