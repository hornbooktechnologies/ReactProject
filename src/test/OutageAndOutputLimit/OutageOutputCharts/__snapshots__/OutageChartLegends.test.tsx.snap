// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: OutageChartLegends OutageChartLegends renders correctly 1`] = `
<div
  className="timeseries-legends"
>
  <div
    className="uk-flex uk-flex-middle uk-flex-center"
    uk-flex=""
  >
    <div
      className="uk-flex uk-flex-middle pl-20 mb-10 ml-10 mr-10"
    >
      <span
        className="indicate actualLegend"
      />
      <div
        className="ml-10"
      >
        <p
          className="uk-margin-remove font-12"
        >
          OUTAGE_AND_OUTPUT_LIMIT.CHART_LEGENDS.ACTUAL
        </p>
      </div>
    </div>
    <div
      className="uk-flex uk-flex-middle pl-20 mb-10 ml-10 mr-10"
    >
      <span
        className="indicate forecastLegend"
      />
      <div
        className="ml-10"
      >
        <p
          className="uk-margin-remove font-12"
        >
          OUTAGE_AND_OUTPUT_LIMIT.CHART_LEGENDS.FORECAST
        </p>
      </div>
    </div>
    <div
      className="uk-flex uk-flex-middle pl-20 mb-10 ml-10 mr-10"
    >
      <span
        className="indicate salesLegend"
      />
      <div
        className="ml-10"
      >
        <p
          className="uk-margin-remove font-12"
        >
          OUTAGE_AND_OUTPUT_LIMIT.CHART_LEGENDS.SALES_UNIT_PRICE
        </p>
      </div>
    </div>
  </div>
</div>
`;
